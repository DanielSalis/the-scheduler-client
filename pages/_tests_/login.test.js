import { shallowMount } from '@vue/test-utils';
import LoginPage from '../login.vue'; // Adjust the path if necessary

// Mock the $cookies dependency
const mockCookies = {
  removeAll: jest.fn(),
  set: jest.fn(),
};

describe('LoginPage.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(LoginPage, {
      mocks: {
        $cookies: mockCookies,
        $axios: {
          post: jest.fn(() => Promise.resolve({ data: 'token' })),
        },
        $router: {
          push: jest.fn(),
        },
      },
    });
  });

  it('matches the snapshot', () => {
    // Create a snapshot of the component
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('removes all cookies before mounting', () => {
    // Check that removeAll was called when the component is mounted
    expect(mockCookies.removeAll).toHaveBeenCalled();
  });

  it('logs in the user and sets a cookie', async () => {
    wrapper.setData({ email: 'test@example.com', password: 'password123' });

    await wrapper.vm.loginUser();

    // Check that the axios post method was called with correct parameters
    expect(wrapper.vm.$axios.post).toHaveBeenCalledWith('/user/auth', {
      email: 'test@example.com',
      password: 'password123',
    });

    // Check that the user is redirected to the home page
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith('/');
  });

  it('handles login failure', async () => {
    wrapper.vm.$axios.post.mockRejectedValueOnce(new Error('Invalid credentials'));

    wrapper.setData({ email: 'test@example.com', password: 'wrongpassword' });

    await wrapper.vm.loginUser();

    // Check that no cookie is set on failure
    expect(mockCookies.set).not.toHaveBeenCalled();

    // Check that the user is not redirected
    expect(wrapper.vm.$router.push).not.toHaveBeenCalled();
  });
});