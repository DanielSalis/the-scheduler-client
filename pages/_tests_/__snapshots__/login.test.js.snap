// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage.vue matches the snapshot 1`] = `
"<div tabindex=\\"0\\" class=\\"admin-page\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" rounded=\\"true\\" class=\\"admin-page__content\\">
    <v-form-stub lazyvalidation=\\"true\\" value=\\"true\\">
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" label=\\"email\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" solo=\\"true\\" type=\\"text\\" required=\\"\\"></v-text-field-stub>
      <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" label=\\"senha\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" solo=\\"true\\" type=\\"password\\" required=\\"\\"></v-text-field-stub>
      <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" block=\\"true\\" type=\\"submit\\" validate-on=\\"submit\\" class=\\"mr-4\\">
        Entrar
      </v-btn-stub>
    </v-form-stub>
  </v-card-stub>
</div>"
`;
